apps/mitt-konto/src/MittKonto/Main/Views/CreditCardUpdateView.purs

```diff
@@ -0,0 +1,194 @@
+module MittKonto.Main.CreditCardUpdateView where
+
+import Prelude
+
+import Bottega (BottegaError, bottegaErrorMessage)
+import Bottega.Models (CreditCard, CreditCardRegister, CreditCardRegisterState(..))
+import Data.Either (Either(..))
+import Data.Maybe (Maybe(..))
+import Effect (Effect)
+import Effect.Aff (Aff)
+import Effect.Aff as Aff
+import Effect.Class (liftEffect)
+import Effect.Exception (error)
+import KSF.AsyncWrapper as AsyncWrapper
+import KSF.CreditCard.Choice (choice) as Choice
+import KSF.CreditCard.Register (register) as Register
+import KSF.Sentry as Sentry
+import KSF.User (PaymentTerminalUrl)
+import KSF.User (getCreditCardRegister, registerCreditCard, updateCreditCardSubscriptions) as User
+import MittKonto.Wrappers (class RouteWrapperContent, AutoClose(..), SetRouteWrapperState)
+import MittKonto.Wrappers.Elements as WrapperElements
+import React.Basic (JSX)
+import React.Basic.Classic (make)
+import React.Basic.Classic as React
+import React.Basic.DOM as DOM
+import Record as Record
+
+type BaseProps =
+  ( creditCards :: Array CreditCard
+  , logger      :: Sentry.Logger
+  )
+
+type Inputs = Record BaseProps
+
+newtype RouteWrapperContentInputs = RouteWrapperContentInputs Inputs
+
+type Props =
+  { setWrapperState :: SetRouteWrapperState
+  | BaseProps
+  }
+
+type Self = React.Self Props State
+
+type State =
+  { asyncWrapperState :: AsyncWrapper.Progress JSX
+  , updateState       :: UpdateState
+  , poller            :: Aff.Fiber Unit
+  }
+
+type SetState = (State -> State) -> Effect Unit
+
+data UpdateState
+  = ChooseCreditCard
+  | RegisterCreditCard PaymentTerminalUrl
+
+creditCardUpdateView :: Props -> JSX
+creditCardUpdateView = make component { initialState, render, didMount }
+
+instance viewWrapperContentCardUpdate :: RouteWrapperContent RouteWrapperContentInputs where
+  instantiate (RouteWrapperContentInputs inputs) setWrapperState = do
+    renderedContent <- pure $ creditCardUpdateView $ Record.merge inputs { setWrapperState }
+    setWrapperState \s -> s { titleText = "Uppdatera ditt kredit- eller bankkort"
+                            , renderedContent = renderedContent
+                            }
+
+initialState :: State
+initialState =
+  { asyncWrapperState: AsyncWrapper.Ready
+  , poller: pure unit
+  , updateState: ChooseCreditCard
+  }
+
+component :: React.Component Props
+component = React.createComponent "CreditCardUpdateView"
+
+didMount :: Self -> Effect Unit
+didMount self@{ setState, props: { creditCards, logger, setWrapperState } } = do
+  setState _ { asyncWrapperState = AsyncWrapper.Ready }
+  Aff.launchAff_ do
+    case creditCards of
+      []       -> liftEffect $ do
+        logger.log "No credit cards found" Sentry.Error
+        onError self
+      [ card ] -> do
+        registerCreditCard self card
+      _        -> pure unit
```

Comment:
Should we take `Maybe CreditCard` (or do we even need `Maybe`?) in props instead of `Array CreditCard` if only one card makes sense in this component?

Reply:
Ohh, you can choose if there are multiple cards, but if there's only one, it's registered automatically?

Reply:
> Ohh, you can choose if there are multiple cards, but if there's only one, it's registered automatically?

Yes that is the deal